spec/factorial_spec.rb

```diff
@@ -0,0 +1,23 @@
+require './solver'
+
+describe Solver do
+  context 'Testing the factorial method' do
+    solver = Solver.new
+
+    it 'factorial 3 should be 6' do
+      expect(solver.factorial(3)).to eq(6)
+    end
+
+    it 'factorial 5 should be 120' do
+      expect(solver.factorial(5)).to eq(120)
+    end
+
+    it 'factorial 0 should be 1' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'factorial 1 should be 1' do
+      expect(solver.factorial(1)).to eq(1)
+    end
```

Comment:
- [ ] Great job so far @mertkantaroglu and @josuerushanika However I noticed that you're missing one of the tests under the `#factorial` test cases. Per the project requirements below, on the last line, there needs to also be a test that handles a negative value which raises an exception as an expectation 😉 
- [ ] So kindly include that in your tests as well 💯 

> Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.

Reply:
Thank you for the heads up @NahnahAJ